packages/plugin-core/src/commands/MergeNoteCommand.ts

```diff
@@ -0,0 +1,453 @@
+    destNote: NoteProps | undefined;
+  }) {
+    const ctx = `${this.key}:prepareProxyMetricPayload`;
+    const { sourceNote, destNote } = opts;
+    if (sourceNote === undefined || destNote === undefined) {
+      // source or dest note undefined, this could be from cancellation.
+      // just return.
+      return;
+    }
+    const sourceBasicStats = StatisticsUtils.getBasicStatsFromNotes([
+      sourceNote,
+    ]);
+    const destBasicStats = StatisticsUtils.getBasicStatsFromNotes([destNote]);
+    if (sourceBasicStats === undefined || destBasicStats === undefined) {
+      this.L.error({ ctx, message: "failed to get basic stats from notes." });
+      return;
+    }
+
+    const { numChildren, numLinks, numChars, noteDepth } = sourceBasicStats;
+    const {
+      numChildren: destNumChildren,
+      numLinks: destNumLinks,
+      numChars: destNumChars,
+      noteDepth: destNoteDepth,
+    } = destBasicStats;
+
+    const sourceTraits = sourceNote.traits;
+    const destTraits = destNote.traits;
+
+    const engine = this.extension.getEngine();
+    this._proxyMetricPayload = {
+      command: this.key,
+      numVaults: engine.vaults.length,
+      numChildren,
+      numLinks,
+      numChars,
+      noteDepth,
+      traits: sourceTraits || [],
+      extra: {
+        destNumChildren,
+        destNumLinks,
+        destNumChars,
+        destNoteDepth,
+        destTraits: destTraits || [],
+      },
+    };
+  }
+
+  /**
+   * Given a source note and destination note,
+   * append the entire body of source note to the destination note.
+   * @param sourceNote Source note
+   * @param destNote Dest note
+   */
+  private async appendNote(opts: {
+    sourceNote: NoteProps;
+    destNote: NoteProps;
+  }) {
+    const { sourceNote, destNote } = opts;
+    // grab body from current active note
+    const appendPayload = sourceNote.body;
+
+    // append to end
+    const destBody = destNote.body;
+    const newBody = `${destBody}\n\n${appendPayload}`;
+    destNote.body = newBody;
+    const writeResp = await this.extension.getEngine().writeNote(destNote);
+    if (!writeResp.error) {
+      return writeResp.data || [];
+    } else {
+      this.L.error(writeResp.error);
+      return [];
+    }
+  }
+
+  /**
+   * Helper for {@link updateBacklinks}.
+   * Given a note id, source and dest note,
+   * Find all links in note with id that points to source
+   * and update it to point to dest instead.
+   * @param opts
+   */
+  private async updateLinkInNote(opts: {
+    id: string;
+    sourceNote: NoteProps;
+    destNote: NoteProps;
+  }) {
+    const ctx = `${this.key}:updateLinkInNote`;
+    const { id, sourceNote, destNote } = opts;
+    const noteToUpdate = await this.extension.getEngine().getNote(id);
+    if (noteToUpdate !== undefined) {
+      const linksToUpdate = noteToUpdate.links.filter(
+        (link) => link.value === sourceNote.fname
+      );
+      const noteToUpdateDeepCopy = _.cloneDeep(noteToUpdate);
```

Comment:
in this case since we are intentionally mutating the note body, do we need to clone?

Reply:
also `getNote` shoudl already clone the note

Reply:
I was being extra to not mutate it before I explicitly call write but I guess this is now taken care of with the recent store / engine improvements 😄